persist-cli/src/main/java/io/ballerina/persist/cmd/Migrate.java

```diff
@@ -0,0 +1,778 @@
+        if (1 < argList.size()) {
+            errStream.println("Error: Too many arguments provided. Only one argument " +
+                    "is allowed to pass for migration label");
+            return;
+        }
+
+        String migrationName = argList.get(0);
+
+        migrate(migrationName, projectPath.toString());
+
+    }
+
+    private static void migrate(String migrationName, String projectDirName) {
+        String balFile = getBalFilePath(projectDirName);
+
+        if (balFile != null) { // todo add validation in execute
+            Path balFilePath = Paths.get(balFile);
+
+            String persistDirPath = projectDirName + "/persist";
+
+            // Create a File object for the persist directory
+            File persistDir = new File(persistDirPath);
+
+            // Create a File object for the migrations directory
+            File migrationsDir = new File(persistDir, "migrations");
+
+            // Check if the migrations directory exists
+            if (!migrationsDir.exists()) {
+                // Create the migrations directory
+                boolean created = migrationsDir.mkdir();
+                if (!created) {
+                    errStream.println("Failed to create migrations directory");
+                    return;
+                }
+
+                // Create a blankFile.bal file that contains the line "boom"
+                Path blankFilePath = Paths.get(persistDirPath + "/blankFile.bal");
+                try {
+                    Files.createFile(blankFilePath);
+                    try (BufferedWriter buffWriter = Files.newBufferedWriter(blankFilePath, StandardCharsets.UTF_8)) {
+                        buffWriter.write("import ballerina/persist as _;\n");
+                    }
+                } catch (IOException e) {
+                    errStream.println("Error: Failed to create blankFile.bal: " + e.getMessage());
+                    return;
+                }
+
+                // Migrate with the blankFile.bal
+                migrateWithTimestamp(migrationsDir, migrationName, balFilePath, blankFilePath);
+
+                // Delete the blankFile.bal
+                try {
+                    Files.delete(blankFilePath);
+                } catch (IOException e) {
+                    errStream.println("Error: Failed to delete blankFile.bal: " + e.getMessage());
+                }
+
+            } else {
+                // Migrate with the latest bal file in the migrations directory
+                migrateWithTimestamp(migrationsDir, migrationName, balFilePath,
+                        findLatestBalFile(getSubfolderNames(migrationsDir.toString())));
+            }
+        }
+    }
+
+    // Get a list of all timestamp folders in the migrations directory
+    private static List<String> getSubfolderNames(String folderPath) {
+        List<String> subfolderNames = new ArrayList<>();
+        File folder = new File(folderPath);
```

Comment:
You don't need to create File object to do path manipulations. there is `Files` and `Paths` apis which provides functions